retrievalmarket/types.go

```diff
@@ -244,6 +256,28 @@ func (p Params) SelectorSpecified() bool {
 	return p.Selector != nil && !bytes.Equal(p.Selector.Raw, cbg.CborNull)
 }
 
+func (p Params) IntervalLowerBound(currentInterval uint64) uint64 {
+	intervalSize := p.PaymentInterval
+	var lowerBound uint64
+	var target uint64
+	for target < currentInterval {
+		lowerBound = target
+		target += intervalSize
+		intervalSize += p.PaymentIntervalIncrease
+	}
+	return lowerBound
+}
+
+func (p Params) NextInterval(currentInterval uint64) uint64 {
```

Comment:
It's easier to keep track of the accounting if we count totals from zero, instead of counting deltas from some interval point - when there's a restart, using deltas gets much more complicated.
So we're now treating the interval as an absolute value from zero, instead of as a delta. eg:
```
Delta intervals:    1000, 1500, 2000
Absolute intervals: 1000, 2500, 4500
```